lib/control-app.js

```diff
@@ -1,139 +1,135 @@
+    fullPaths: true
+  })
 
-    // default builder is browserify which we provide
-    config.builder = config.builder || defaultBuilder;
+  files = deglob.sync(files, {cwd: prj_dir})
 
-    var build = require(config.builder)(files, config);
+  var user_html = ''
+  if (config.html) {
+    user_html = fs.readFileSync(path.join(prj_dir, config.html), 'utf-8')
+  }
 
-    var app = express();
-    app.use(compression());
+  // default builder is browserify which we provide
+  config.builder = config.builder || defaultBuilder
 
-    expstate.extend(app);
+  var build = require(config.builder)(files, config)
 
-    app.set('state namespace', 'zuul');
-    app.expose(ui, 'ui');
-    app.expose(config.name, 'title');
+  var app = express()
+  app.use(compression())
 
-    app.set('views', __dirname + '/../frameworks');
-    app.set('view engine', 'html');
-    app.engine('html', require('hbs').__express);
+  expstate.extend(app)
 
-    app.use(function(req, res, next) {
-        res.locals.title = config.name;
-        res.locals.user_scripts = config.scripts || [];
-        res.locals.user_html = user_html;
-        next();
-    });
+  app.set('state namespace', 'zuul')
+  app.expose(ui, 'ui')
+  app.expose(config.name, 'title')
 
+  app.set('views', __dirname + '/../frameworks')
+  app.set('view engine', 'html')
+  app.engine('html', require('hbs').__express)
 
-    app.get('/__zuul', function(req, res) {
-        res.locals.config = { port: config.support_port };
-        res.render('index');
-    });
+  app.use(function (req, res, next) {
+    res.locals.title = config.name
+    res.locals.user_scripts = config.scripts || []
+    res.locals.user_html = user_html
+    next()
+  })
 
-    var bundle_router = new express.Router();
-    app.use(bundle_router);
+  app.get('/__zuul', function (req, res) {
+    res.locals.config = { port: config.support_port }
+    res.render('index')
+  })
 
-    // airtap files
-    app.use('/__zuul', express.static(__dirname + '/../frameworks'));
-    // framework files
-    app.use('/__zuul', express.static(framework_dir));
+  var bundle_router = new express.Router()
+  app.use(bundle_router)
 
-    // any user's files
-    app.use(express.static(process.cwd()));
+  // airtap files
+  app.use('/__zuul', express.static(__dirname + '/../frameworks'))
 
-    if (config.coverage && config.local) {
-        // coverage endpoint
-        app.use('/__zuul/coverage', im.createHandler());
-    }
+  // any user's files
+  app.use(express.static(process.cwd()))
 
-    var map = undefined;
+  if (config.coverage && config.local) {
+    // coverage endpoint
+    app.use('/__zuul/coverage', im.createHandler())
+  }
 
-    var clientBundler = browserify(opt);
-    clientBundler.require(path.join(framework_dir, '/client.js'), { entry: true });
+  var map = undefined
+  var clientBundler = browserify(opt)
+  var tape = path.resolve(__dirname, '../frameworks/tape.js')
+  clientBundler.require(tape, { entry: true })
```

Comment:
Do we still need the `.require`, which makes tape available from outside the bundle? Can perhaps simplify it to:

```js
var tape = path.resolve(__dirname, '../frameworks/tape.js')
var clientBundler = browserify(tape, opt)
```

Reply:
Works well!

Reply:
Btw, if we are to remove the user interface we should just cache this bundle in memory and not bundle at request.

Reply:
Then we could also get rid of `watchify`

Reply:
There's a use case for watchify though? Not having to restart the server during development.

Reply:
Yep. I was thinking that if you are testing on many browsers (not in development) you want this to be fast. Just thinking out loud.

Reply:
That's true. So that's another case for a CI-only option (like `loopback`).